### PR TITLE
Combine publish-docs.sh and release.sh

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
publish-docs.sh expects to only run on the main branch.
However we also want to generate javadocs on branches
other than main, for example to publish updated deprecation
notices.

Modifications:
- Remove publish-docs.sh, and move its logic into release.sh

Result:
Non-main branches will have javadocs published during release.

### CODE REVIEW

**Comment by Scottmitch** [comment]
Depends upon https://github.com/apple/servicetalk/pull/2195

File: scripts/release.sh [hunk 1]
```diff
@@ -67,6 +85,9 @@ if ( echo "$version" | grep -q "SNAPSHOT" ); then
     exit 1
 fi
 
+# Clean up the state at the beginning in case the previous run did not finish successfully
+clean_up_gh_pages
+
 if [ -z "${DRYRUN:-}" ]; then
     gradle_build_args="--no-build-cache --warning-mode all --refresh-dependencies clean build publishToMavenLocal"
```
Comment (idelpivnitskiy) [inline scripts/release.sh:92]: I've left a comment how to simplify this command on the PR that is already merged, we can update it here: https://github.com/apple/servicetalk/pull/2193#discussion_r854283209

**Review by idelpivnitskiy — APPROVED** [review]

File: scripts/release.sh (hunk 1 above)
Comment (Scottmitch) [reply scripts/release.sh]: used `assemble` as recommended.

**Comment by Scottmitch** [comment]
0.41 (7f4e63349)

**Pull request merged**